parsons/databases/redshift/redshift.py

```diff
@@ -420,32 +424,66 @@ def copy(self, table_obj, table_name, if_exists='fail', max_errors=0, distkey=No
                 See :ref:`parsons-table` for output options.
         """
 
-        # To Do:
-        # Auto split big files to copy more quickly
-        # Consider a json for better stability
-
-        key = self.temp_s3_copy(table_obj)
-
-        cols = None
+        # Specify the columns for a copy statement.
         if specifycols:
-            cols = table_obj.columns
+            cols = tbl.columns
+        else:
+            cols = None
 
-        try:
+        key = None
 
-            self.copy_s3(table_name, self.s3_temp_bucket, key,
-                         if_exists=if_exists, max_errors=max_errors,
-                         distkey=distkey, sortkey=sortkey, padding=padding, ignoreheader=1,
-                         varchar_max=varchar_max, statupdate=statupdate, compupdate=compupdate,
-                         acceptanydate=acceptanydate, dateformat=dateformat, timeformat=timeformat,
-                         blanksasnull=blanksasnull, nullas=nullas, emptyasnull=emptyasnull,
-                         acceptinvchars=acceptinvchars, truncatecolumns=truncatecolumns,
-                         columntypes=columntypes, specifycols=cols,
-                         aws_access_key_id=aws_access_key_id,
-                         aws_secret_access_key=aws_secret_access_key, compression='gzip')
+        with self.connection() as connection:
 
-        finally:
+            try:
+                # Check to see if the table exists. If it does not or if_exists = drop, then
+                # create the new table.
+                if self._create_table_precheck(connection, table_name, if_exists):
+                    sql = self.create_statement(tbl, table_name, padding=padding,
+                                                distkey=distkey, sortkey=sortkey,
+                                                varchar_max=varchar_max,
+                                                columntypes=columntypes)
+                    self.query_with_connection(sql, connection, commit=False)
+                    logger.info(f'{table_name} created.')
+
+                # If alter_table is True, then alter table if the table column widths
+                # are wider than the existing table.
+                elif alter_table:
+                    self.alter_varchar_column_widths(tbl, table_name)
 
-            self.temp_s3_delete(key)
+                else:
+                    pass
```

Comment:
You don't need this `else: pass` here.. I feel like the code reads a little better without it.